extensions/vscode/src/utils/names.ts

```diff
@@ -82,3 +52,45 @@ export function contentRecordNameValidator(
     return undefined;
   };
 }
+
+/**
+ * Creates a semi-unique configuration name from a content title.
+ *
+ * @param title The title of the content to create a filename from
+ * @param existingNames [[]] An array of existing configuration names to ensure
+ *   uniqueness
+ * @returns A filename that is safe to use in the filesystem with a unique 4
+ * character ending to avoid Git conflicts.
+ */
+export function newConfigFileNameFromTitle(
+  title: string,
+  existingNames: string[] = [],
+): string {
+  const filename = filenamify(title, {
+    replacement: "-",
+    maxLength: 95,
+  });
+
+  // Generate unique name endings until we find a unique one
+  let result;
+  do {
+    const uniqueEnding = randomNameEnding(4);
+    result = `${filename}-${uniqueEnding}`;
```

Comment:
```suggestion
    result = `${filename}:${uniqueEnding}`;
```
Another nit... a colon rather than a dash

Reply:
This one feels weird to me. I almost never use colons in my file names personally. Not saying I'm against it just voicing my immediate reaction.